docs/resolver-template/cmd/demoresolver/main.go

```diff
@@ -51,6 +53,22 @@ func (r *resolver) GetSelector(context.Context) map[string]string {
 	}
 }
 
+// ValidateRequest ensures the ResolutionSpec from a request is as expected.
+// This is currently at an ALPHA stability level and subject to alpha API compatibility policies.
+func (r *resolver) ValidateRequest(ctx context.Context, req *v1beta1.ResolutionRequestSpec) error {
+	if len(req.Params) > 0 {
+		return r.ValidateParams(ctx, req.Params)
+	}
+	u, err := neturl.ParseRequestURI(req.URL)
+	if err != nil {
+		return err
+	}
```

Comment:
I wonder if this part of the logic is common to all implementations and should be done by the framework instead.
The `ValidateRequests` could take a parsed URL as input instead.

Reply:
Oh. I just realized. The reason for passing the ResolutionSpec as an input is that if we add new fields in the future, we don't need to add a yet another function to stay compatible with the API. Leaving the spec as an input means that the signature of the function does not need to change.